File: java/GateFour.java/Level6.java
 public class Level6{

    public int[][] splitEvenOdd(int [] numbers){
        int countEven = 0;
        int countOdd = 0;
        for(int number : numbers){
         if (number % 2 == 0){
            countEven++;
        
          }

           else{
                countOdd++;       
                }
            }
        int [] even = new int[countEven];
        int [] odd = new int [countOdd];
        
        int evenIndex = 0;
        int oddIndex = 0;
        for(int number : numbers){
             if (number % 2 == 0)
                even[evenIndex++] = number;
            else 
                odd[oddIndex++] = number;
}
            return new int[][]{odd, even};

        }
         


    public boolean isPalindrome(int [] numbers) {

    int left = 0;
    int right = numbers.length - 1;

    while(left < right) {
        if (numbers[left] != numbers[right])
            return false;
                left++;
                right--;

    }
        
        return true;

    }



    public int [] perfectSquares(int [] numbers){
                int size = 0;
                for (int number: numbers){
                    int root = (int)Math.sqrt(number);
                    if(root * root == number)
                        size++;
              }
            int [] result = new int [size]; 
           int index = 0;
             for (int number: numbers){
                int root = (int)Math.sqrt(number);
                    if(root * root == number)
                        result[index++] = number;
        }
            return result;
}

     public int [] replaceNonPerfectSquares(int [] numbers){
        for(int index = 0; index < numbers.length; index++){
             int root = (int)Math.sqrt(numbers[index]);
                    if(root * root != numbers[index]){
                        numbers[index] = -1;

                }
               }

            return numbers;

        }




}

File: python/GateFour/level_6.py
import math
def split_even_odd(numbers):
    odd = []
    even = []
    for number in numbers:
        if number % 2 == 0:
            even.append(number)
        else:
            odd.append(number)
    return[odd, even]

def is_palindrome(numbers):
    left = 0
    right = len(numbers) - 1
    while left < right:
        if numbers[left] != numbers[right]:
            return False
        left += 1
        right -= 1
    return True

def perfect_squares(numbers):
    result = []
    for number in numbers:
        if int(math.sqrt(number))**2 == number:
            result.append(number)
    return result

def replace_non_perfect(numbers):
    for index in range(len(numbers)):
        if int(math.sqrt(numbers[index]))**2 != numbers[index]:
            numbers[index] = -1
    return numbers

        
